cprover_bindings/src/goto_program/typ.rs

```diff
@@ -1239,7 +1250,63 @@ impl Type {
         } else if self.is_pointer() {
             Expr::pointer_constant(0, self.clone())
         } else {
-            unreachable!("Can't convert {:?} to a one value", self);
+            unreachable!("Can't convert {:?} to a zero value", self);
+        }
+    }
+
+    pub fn zero_initializer(&self, st: &SymbolTable) -> Expr {
+        let concrete = self.unwrap_typedef();
+        match concrete {
+            // Base case
+            Bool
+            | CBitField { .. }
+            | CInteger(_)
+            | Double
+            | Float
+            | Pointer { .. }
+            | Signedbv { .. }
+            | Unsignedbv { .. } => self.zero(),
+
+            // Recursive cases
+            Array { typ, size } => typ.zero_initializer(st).array_constant(*size),
+            InfiniteArray { typ } => typ.zero_initializer(st).infinite_array_constant(),
+            Struct { components, .. } => {
+                let values: Vec<Expr> =
+                    components.iter().map(|c| c.typ().zero_initializer(st)).collect();
+                Expr::struct_expr_from_padded_values(self.clone(), values, st)
+            }
+            StructTag(tag) => st.lookup(*tag).unwrap().typ.zero_initializer(st),
+            TypeDef { .. } => unreachable!("Should have been normalized away"),
+            Union { components, .. } => {
+                if components.is_empty() {
+                    Expr::empty_union(self.clone(), st)
+                } else {
+                    let largest = components.iter().max_by_key(|c| c.sizeof_in_bits(st)).unwrap();
```

Comment:
Nit: This could be written in a nicer way with `if let Some(largest) = ... { ... } else { /* empty case */ }`, thus avoiding unwrap().